web/web/static/web/js/components/pane.js

```diff
@@ -0,0 +1,1688 @@
+     */
+    self.back = function() {
+        self.onBack({});
+    };
+
+    /**
+     * Register a diff event.
+     */
+    self.diff = function() {
+        self.onDiff({});
+    };
+
+    /**
+     * Register a clone event.
+     */
+    self.clone = function() {
+        self.onClone({});
+    };
+
+    /**
+     * Register a close event.
+     */
+    self.close = function() {
+        self.onClose({});
+    };
+}
+
+angular.module('cloudSnitch').component('panetopctrl', {
+    templateUrl: '/static/web/html/panes/panetopctrl.html',
+    controller: [PaneTopCtrlController],
+    bindings: {
+        backable: '<',
+        cloneable: '<',
+        diffable: '<',
+        onClose: '&',
+        onClone: '&',
+        onDiff: '&',
+        onBack: '&'
+    }
+});
+
+/**
+ * Controller for the panes component.
+ *  Handles multiple panes and directs pane to pane communication.
+ */
+function PanesController(typesService, timeService) {
+    var self = this;
+
+    self.$onInit = function() {
+        self.maxPanes = self.maxPanes || 2;
+        self.panes = [];
+        self.diff = undefined;
+
+        // Start with one pane.
+        self.add();
+    };
+
+    /**
+     * Add a pane
+     */
+    self.add = function() {
+        // Return early if adding another pane would exceed max panes
+        if (self.panes.length >= self.maxPanes) {
+            return true;
+        }
+        self.panes.push({});
+    };
+
+    /**
+     * Close a pane.
+     */
+    self.close = function(index) {
+        if (index < self.panes.length) {
+            self.panes.splice(index, 1);
+        }
+    };
+
+    /**
+     * Clone a pane
+     */
+    self.clone = function(frames) {
+        if (self.panes.length < self.maxPanes) {
+            self.panes.push({frames: frames});
+        }
+    };
+
+    /**
+     * Handle a pane change. example search -> results -> details
+     */
+    self.paneChange = function(index, frame) {
+        self.panes[index].topFrame = frame;
+    };
+
+    /**
+     * Determine cloneability
+     */
+    self.cloneable = function() {
+        return self.panes.length < self.maxPanes;
+    };
+
+    /**
+     * Determine if panes are diffable.
+     * They must be the same type and on details with the same id.
+     * @TODO - disallow same times.
+     */
+    self.diffable = function() {
+        if (self.panes.length > 1) {
+            var a = self.panes[0].topFrame;
+            var b = self.panes[1].topFrame;
+
+            if (a === undefined || b === undefined) {
+                return false;
+            }
+
+            if (a.state != 'details' || b.state != 'details') {
+                return false;
+            }
+
+            if (a.type != b.type) {
+                return false;
+            }
+
+            var aId = a.record[a.type][typesService.identityProperty(a.type)];
+            var bId = b.record[b.type][typesService.identityProperty(b.type)];
+            if (aId != bId) {
```

Comment:
don't necessarily think this is wrong but why do the ID's need to match? we can't diff across environments? or is that not the ID you're checking here

Reply:
That is correct, we do not support diffing across environments at this time.